src/wp-includes/class-wp-scripts.php

```diff
@@ -714,6 +781,262 @@ public function in_default_dir( $src ) {
+	 * @return bool True on success, false on failure.
+	 */
+	public function add_data( $handle, $key, $value ) {
+		if ( 'script_args' === $key ) {
+			$args = $this->get_normalized_script_args( $handle, $value );
+			if ( $args['in_footer'] ) {
+				parent::add_data( $handle, 'group', 1 );
+			}
+			return parent::add_data( $handle, $key, $args );
+		}
+		return parent::add_data( $handle, $key, $value );
+	}
+
+	/**
+	 * Checks all handles for any delayed inline scripts.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @return bool True if the inline script present, otherwise false.
+	 */
+	public function has_delayed_inline_script() {
+		foreach ( $this->registered as $handle => $script ) {
+			// Non standalone scripts in the after position, of type async or defer, are usually delayed.
+			$strategy = $this->get_intended_strategy( $handle );
+			if ( $this->is_non_blocking_strategy( $strategy ) &&
+				$this->has_non_standalone_inline_script( $handle, 'after' )
+			) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Normalize the data inside the $args parameter and support backward compatibility.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string        $handle Name of the script.
+	 * @param array         $args     {
+	 *      Optional. Additional script arguments. Default empty array.
+	 *
+	 *      @type boolean   $in_footer    Optional. Default false.
+	 *      @type string    $strategy     Optional. Values blocking|defer|async. Default 'blocking'.
+	 * }
+	 * @return array        Normalized $args array.
+	 */
+	private function get_normalized_script_args( $handle, $args = array() ) {
+		$default_args = array(
+			'in_footer' => false,
+			'strategy'  => 'blocking',
+		);
+
+		// Handle backward compatibility for $in_footer.
+		if ( true === $args ) {
+			$args = array( 'in_footer' => true );
+		}
+
+		return wp_parse_args( $args, $default_args );
+	}
+
+	/**
+	 * Get all dependents of a script.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $handle The script handle.
+	 * @return array Array of script handles.
+	 */
+	private function get_dependents( $handle ) {
+		// Check if dependents map for the handle in question is present. If so, use it.
+		if ( array_key_exists( $handle, $this->dependents_map ) ) {
+			return $this->dependents_map[ $handle ];
+		}
+
+		$dependents = array();
+
+		// Iterate over all registered scripts, finding dependents of the script passed to this method.
+		foreach ( $this->registered as $registered_handle => $args ) {
+			if ( in_array( $handle, $args->deps, true ) ) {
+				$dependents[] = $registered_handle;
+			}
+		}
+
+		// Add the handles dependents to the map to ease future lookups.
+		$this->dependents_map[ $handle ] = $dependents;
+
+		return $dependents;
+	}
+
+	/**
+	 * Determines if a script loading strategy is valid.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $strategy A script loading strategy.
+	 * @return bool True if the strategy is of an allowed type, false otherwise.
```

Comment:
```suggestion
	 * @return bool True if the strategy is valid, false otherwise.
```
The previous description was tightly coupled to implementation. This change suggests keeping the return value description consistent with the method's name and description.

Reply:
@costdev resolved in a recent commit, thank you.